seed/seed.go

```diff
@@ -46,6 +48,13 @@ func New(pp ...Param) *Seeder {
 // Seed the provided config with values for their sources.
 func (s *Seeder) Seed(cfg *config.Config) error {
 	seedMap := make(map[*config.Field]bool, len(cfg.Fields))
+	flagset := flag.NewFlagSet("Harvester flags", flag.ContinueOnError)
```

Comment:
It would be cleaner if the flag seeding logic was moved to a `seed/flag` package with a `Getter` implementation like `seed/consul`

Reply:
I totally agree this would be a better approach, though the way flags behave is different from how the current `Getter` implementation works as we can't simply do a `Get(key)`: we first need to tell what flags are expected, and only when they are all defined, we can do a `Parse()` and then `Get(key)`.

The reason why I didn't implement it this way is because I can't think of any easy/concise implementation for this PR.

What I can think of, which is more like refactoring and IMO could belong to another ticket, is to actually create one object per seed type (env, flag, consul, etc.) implementing their own `Seed(cfg *config.Fields) error` method, so that they can handle values however they want, without depending on the pre-existing `for` loop.

What do you think?

Reply:
The flag parsing could take place in flag specifc `Getter` implementation constructor and `Get(key)` will return the appropriate value.
I agree, though, that this should better be handled in different issue that might refactor existing entities too.